### PR TITLE
feat!(ENG-1662): kill drain grpc. deactivates drains first.

### PR DESCRIPTION
**Pull request by itayd**
- kill drain endpoint. deactivate will always drain first.
- auto deactivate deployment once all sessions are done.

### CODE REVIEW

**Review by daabr — APPROVED** [review]
I'm uncomfortable with the non-atomic nature of the `DeploymentHasActiveSessions` function. We can have sessions start between line 225 and line 229, and switch to "running" state between lines 212 and 220, i.e. while this function is running.

But I don't have a way to solve this, because we can't globally lock sessions every time this function is running. WDYT?

**Comment by itayd** [comment]
> I'm uncomfortable with the non-atomic nature of the `DeploymentHasActiveSessions` function. We can have sessions start between line 225 and line 229, and switch to "running" state between lines 212 and 220, i.e. while this function is running.
> 
> But I don't have a way to solve this, because we can't globally lock sessions every time this function is running. WDYT?

This is always happening always in a transaction.

File: internal/backend/db/dbgorm/deployments.go [hunk 1]
```diff
@@ -201,3 +202,29 @@ func (db *gormdb) ListDeployments(ctx context.Context, filter sdkservices.ListDe
 		return kittehs.TransformError(ds, scheme.ParseDeployment)
 	}
 }
+
+func (db *gormdb) DeploymentHasActiveSessions(ctx context.Context, id sdktypes.DeploymentID) (bool, error) {
+	r, err := db.ListSessions(ctx, sdkservices.ListSessionsFilter{
+		DeploymentID: id,
+		StateType:    sdktypes.SessionStateTypeRunning,
```
Comment (ig-ra) [inline internal/backend/db/dbgorm/deployments.go]: not for this PR of coz, but maybe ListSessions could receive several states somehow? e.g as a list `Running, Created`? or bit mask: `Running | Created`?

Comment (itayd) [reply internal/backend/db/dbgorm/deployments.go]: maybe in the future.

**Pull request merged**